### PR TITLE
Allow user to save an image link

### PR DESCRIPTION
**Pull request by ZhuonanLin**
Closes #2

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by ZhuonanLin** [comment]
[Issue 2: Save Image Link](https://trello.com/c/TqUkAPVg/103-issue-2-save-image-link)

File: test/models/image_test.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  def test_link_validation__http
+    image = Image.new(link: 'http://www.google.com')
+    assert_predicate image, :valid?
+  end
+
+  def test_link_validation__invalid_link
+    image = Image.new(link: 'asdfas')
+    # refute_predicate image, :valid?
+    assert_equal false, image.valid?
```
[the remark this answers is not in the document]
Comment (ZhuonanLin) [reply test/models/image_test.rb]: Hi, I tried to use the `refute_predicate`, but the Rubocop complains about it and suggests to use `assert_not_predicate` instead.

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+class ImagesCotrollerTest < ActionDispatch::IntegrationTest
+  def test_new
+    get new_image_path
+    # puts response.body
+    assert_response :ok
+    assert_select 'h1', text: "Let's submit an image url!"
+  end
+
+end
```
[the remark this answers is not in the document]
Comment (ZhuonanLin) [reply test/controllers/images_controller_test.rb]: Addressed.

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require 'test_helper'
+
+class ImagesCotrollerTest < ActionDispatch::IntegrationTest
+  def test_new
+    get new_image_path
+    # puts response.body
+    assert_response :ok
+    assert_select 'h1', text: "Let's submit an image url!"
+  end
+
```
[the remark this answers is not in the document]
Comment (ZhuonanLin) [reply test/controllers/images_controller_test.rb]: This has been fixed in a later commit.

File: app/controllers/images_controller.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class ImagesController < ApplicationController
+  def new
+    @image = Image.new
+  end
+
+end
```
[the remark this answers is not in the document]
Comment (ZhuonanLin) [reply app/controllers/images_controller.rb]: This has been fixed in a later commit.

File: app/views/application/home.html.erb [hunk 5]
```diff
@@ -1 +1,2 @@
 <h1> Welcome! </h1>
+<%= link_to "Submit a new image!",  new_image_path%>
```
Comment (mickiehsu) [inline app/views/application/home.html.erb]: no need to change in this PR, there should be an space after `new_image_path`

**Pull request merged**